geom/alg_convex_hull.go

```diff
@@ -195,3 +180,36 @@ func distanceSq(p, q XY) Scalar {
 	pSubQ := p.Sub(q)
 	return pSubQ.Dot(pSubQ)
 }
+
+func collinearLine(pts []XY) (Geometry, bool) {
+	if len(pts) < 2 {
+		return nil, false
+	}
+	ps := make([]XY, len(pts))
```

Comment:
What's the reason to make a copy of the input? Even if their order is modified, I don't think that would matter for the caller.